### PR TITLE
Feature/14570 integrate audit service

### PR DESCRIPTION
**Pull request by GuyHarwood**
integrated audit service hooks within components.

### CODE REVIEW

File: pupil-spa/src/app/question/question.component.spec.ts [hunk 1]
```diff
@@ -18,7 +25,9 @@ describe('QuestionComponent', () => {
     component = fixture.componentInstance;
     fixture.detectChanges();
     // prevent Timeout's being set
-    spyOn(component, 'ngAfterViewInit').and.returnValue(null);
+    // NOTE - calling this makes no difference to sendTimeoutEvent test.
+    // all other tests work without the call below.
+    //spyOn(component, 'ngAfterViewInit').and.returnValue(null);
```
Comment (GuyHarwood) [inline pupil-spa/src/app/question/question.component.spec.ts]: @jon-shipley i checked every test with a false=true call and none of them failed without it, apart from the audit ones, which rely on ngAfterViewInit

File: pupil-spa/src/app/question/question.component.spec.ts [hunk 2]
```diff
@@ -82,10 +90,12 @@ describe('QuestionComponent', () => {
   });
 
   describe('sendTimeoutEvent', () => {
+    // ISSUE: assertions never run!
```
Comment (GuyHarwood) [inline pupil-spa/src/app/question/question.component.spec.ts]: this assertion never runs, regardless of the ngAfterViewInit faking

File: pupil-spa/src/app/instructions/instructions.component.spec.ts [hunk 3]
```diff
@@ -1,14 +1,18 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-import {Router} from '@angular/router';
+import { Router } from '@angular/router';
 
 import { InstructionsComponent } from './instructions.component';
 import { QuestionServiceMock } from '../question.service.mock';
 import { QuestionService } from '../question.service';
+import { AuditServiceMock } from "../audit.service.mock";
```
Comment (jon-shipley) [inline pupil-spa/src/app/instructions/instructions.component.spec.ts]: Should be single quotes

Comment (GuyHarwood) [inline pupil-spa/src/app/instructions/instructions.component.spec.ts]: good spot, i haven't run the linter! 😱

File: pupil-spa/src/app/instructions/instructions.component.ts [hunk 4]
```diff
@@ -2,6 +2,8 @@ import { Component, OnInit } from '@angular/core';
 import { Router } from '@angular/router';
 
 import { QuestionService } from '../question.service';
+import { AuditService } from "../audit.service";
```
Comment (jon-shipley) [inline pupil-spa/src/app/instructions/instructions.component.ts]: Should be single quotes

File: pupil-spa/src/app/question/question.component.spec.ts (hunk 1 above)
Comment (jon-shipley) [inline pupil-spa/src/app/question/question.component.spec.ts]: If you run the tests in a browser, e.g. Chrome, with the console open, I suspect that you will see timeout's happening in the background (if you configure suitable logging).  It was only in there to shut these down.

File: pupil-spa/src/app/question/question.component.spec.ts (hunk 2 above)
Comment (jon-shipley) [inline pupil-spa/src/app/question/question.component.spec.ts]: Can you leave a TODO item - or xit() the test for later review please?

**Comment by GuyHarwood** [comment]
@jon-shipley linting errors fixed

File: pupil-spa/src/app/question/question.component.spec.ts (hunk 2 above)
Comment (GuyHarwood) [inline pupil-spa/src/app/question/question.component.spec.ts]: 👍

**Comment by jon-shipley** [comment]
😃

File: pupil-spa/src/app/auditEntry.ts [hunk 5]
```diff
@@ -1,9 +1,33 @@
-export type AuditEntryType = 'CheckStarted' | 'QuestionRendered' | 'QuestionAnswered' | 'PauseRendered' | 'UserInput';
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/auditEntry.ts]: do you think it would be better to have one class per file instead of having multiple classes within one?

Comment (GuyHarwood) [inline pupil-spa/src/app/auditEntry.ts]: @GeorgeChatzigiannis it is getting cluttered, but did this to avoid multiple import statements.

**Pull request merged**